### PR TITLE
Bug fix infinite retry case

### PR DESCRIPTION
**Pull request by yorinasub17**
This fixes a bug in the cleanup routine that did not handle successfully detached network interfaces correctly. The bug was due to the ordering of the cases.

Specifically, the original implementation had a function named `isNIDetachedErr` to check if the NI was already detached. Because of the way it was named, it was placed in the secondary switch case after the `err==nil` check. However, in reality the function was checking if `err==nil` and checking the status. But because the `err==nil` check to retry the loop was earlier in the `switch` statement, you would never enter the `isNIDetachedErr` case when `err==nil`, making that an impossible case to satisfy.

To make it easier to follow, this refactors the switch statement to distinctly call out that `isNIDetached` function should be called when `err==nil`. Otherwise, as implemented, it is too easy for someone looking at this code in isolation to "optimize" the switch case and bring the `err==nil` case first.

### CODE REVIEW

**Comment by yorinasub17** [comment]
For reference, we ran into this in the service catalog test: https://app.circleci.com/pipelines/github/gruntwork-io/aws-service-catalog/3112/workflows/213b5fef-f908-424b-95f0-71dab06742d3/jobs/3356

**Comment by yorinasub17** [comment]
UPDATE: Added some basic unit tests for the functionality so we don't have to always go through EKS - 6326fd6

File: .circleci/config.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ defaults: &defaults
     GRUNTWORK_INSTALLER_VERSION: v0.0.21
     TERRATEST_LOG_PARSER_VERSION: v0.29.0
     MODULE_CI_VERSION: v0.28.3
-    TERRAFORM_VERSION: NONE
+    TERRAFORM_VERSION: 0.13.4
```
Comment (rhoboat) [inline .circleci/config.yml:9]: I think we could bump terraform in here and in terraform-aws-eks.

```suggestion
    TERRAFORM_VERSION: 0.13.5
```

File: eks/cleanup.go [hunk 2]
```diff
@@ -270,18 +270,25 @@ func waitForNetworkInterfacesToBeDetached(
...
 			case err == nil:
 				if niResult.Attachment != nil {
 					logger.Warnf("Network interface %s attachment status: %s", aws.StringValue(ni.NetworkInterfaceId), aws.StringValue(niResult.Attachment.Status))
 				}
 				return errors.WithStackTrace(fmt.Errorf("Network Interface %s not detached.", aws.StringValue(ni.NetworkInterfaceId))) // continue retrying
 
-			// Yay, we're detached, process the next network interface.
-			case isNIDetachedErr(niResult, err) || isNINotFoundErr(err):
-				logger.Infof("Network interface %s is detached.", aws.StringValue(ni.NetworkInterfaceId))
+			// If the NI can not be found, then it is already deleted so halt the loop and move on to the next NI.
```
Comment (rhoboat) [inline eks/cleanup.go:286]: NIT: cannot is also a word. 😬 

```suggestion
			// If the NI cannot be found, then it is already deleted so halt the loop and move on to the next NI.
```

**Review by rhoboat — DISMISSED** [review]
Minor nits, but this is great. I'll open another PR in terraform-aws-eks for consuming this.

File: eks/cleanup.go (hunk 2 above)
Comment (yorinasub17) [reply eks/cleanup.go]: TIL!

**Review by rhoboat — APPROVED** [review]
LGTM!

**Pull request merged**